### PR TITLE
Correct Windows builds using the wrong FD_SETSIZE

### PR DESCRIPTION
**Pull request by dgenr8**
https://github.com/BitcoinUnlimited/BitcoinUnlimited/pull/1047

### CODE REVIEW

File: src/util.cpp [hunk 1]
```diff
@@ -671,7 +673,7 @@ bool TruncateFile(FILE *file, unsigned int length) {
  */
 int RaiseFileDescriptorLimit(int nMinFD) {
 #if defined(WIN32)
-    return 2048;
```
Comment (dagurval) [inline src/util.cpp]: So we don't break this again, how about:
```
static_assert(FD_SETSIZE > 64, "compat.h needs to be included first, see PR#406");
```
?

File: src/net.cpp [hunk 2]
```diff
@@ -7,6 +7,10 @@
 #include "config/bitcoin-config.h"
 #endif
 
+// must include first to ensure FD_SETSIZE is correctly set
+// otherwise the default of 64 may be used on Windows
```
Comment (dagurval) [inline src/net.cpp]: Suggest rewording to 
```
// compat.h must be included first to correctly define FD_SETSIZE
```
This is more explicit on what must be included, less unnecessarily details. Details about overriding the value 64 can go into compat if we care. Also, 64 is only default on mingw, not MSVC.

**Comment by dgenr8** [comment]
Pushed changes suggested by @dagurval.

**Comment by dagurval** [comment]
Thanks!
